svm.cpp

```diff
@@ -3043,6 +3043,16 @@ void svm_destroy_param(svm_parameter* param)
 	free(param->weight);
 }
 
+void svm_destroy_problem(svm_problem *problem)
+{
+    delete [] problem->y;
```

Comment:
This assumes that `problem->y` was allocated with `new double[...];`, which is IMO too strong an assumption, especially for a library that exposes a C API.  Since callers are responsible for allocating the `svm_problem` internals, I think it makes sense they'd be responsible for freeing them too.